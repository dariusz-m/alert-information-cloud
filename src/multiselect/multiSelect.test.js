import React from "react";
import ReactTestUtils from "react-dom/test-utils";
import renderer from "react-test-renderer";

import MultiSelect from './multiSelect';

const options = [
    {id: 1, text: "1"},
    {id: 5, text: "Text 5"},
    {text: 'Title', children: [{id: 2, text: "Text 2"}, {id: 3, text: "3"}]},
    {text: 'Title 2', children: [{id: 7, text: "7"}, {id: 8, text: "8"}]}
];

it('Component renders properly.', () => {
    const component = renderer.create(
        <MultiSelect options={options} selectedOptionsIds={[1, 7]} updateSelectedOptions={()=>{}}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
});


it('Options have been properly divided for excluded and included options', () => {
    // Given
    const expectedExcludedOptions = [
        {id: 1, text: "1"},
        {id: 5, text: "Text 5"},
        {text: 'Title', children: []},
        {text: 'Title 2', children: [{id: 8, text: "8"}]}
    ];
    const expectedIncludedOptions = [
        {text: 'Title', children: [{id: 2, text: "Text 2"}, {id: 3, text: "3"}]},
        {text: 'Title 2', children: [{id: 7, text: "7"}]}
    ];
    const component = ReactTestUtils.renderIntoDocument(
        <MultiSelect options={options} selectedOptionsIds={[2, 3, 7]} updateSelectedOptions={()=>{}}/>
    );

    // When
    const {excludedOptions, includedOptions} = component._prepareExcludedAndIncludedOptions();

    // Then
    expect(excludedOptions).toEqual(expectedExcludedOptions);
    expect(includedOptions).toEqual(expectedIncludedOptions);
});
