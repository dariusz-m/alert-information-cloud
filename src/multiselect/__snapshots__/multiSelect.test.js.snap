// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly. 1`] = `
<div
  className="multi-select"
>
  <div
    className="options"
  >
    <div>
      <header>
        AVAILABLE USERS
      </header>
      <input
        className="filter-input"
        onChange={[Function]}
        placeholder="Search..."
      />
      <div
        className="select-all-visible"
      >
        <a
          onClick={[Function]}
        >
          Send to all
        </a>
      </div>
      <ul
        className="ms-list"
      >
        <li
          className=""
          onClick={[Function]}
        >
          Text 5
        </li>
        <li
          className="group-label"
          onClick={[Function]}
        >
          Title
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Text 2
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          3
        </li>
        <li
          className="group-label"
          onClick={[Function]}
        >
          Title 2
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          8
        </li>
      </ul>
    </div>
  </div>
  <div
    className="selected-options"
  >
    <div>
      <header>
        SEND TO
      </header>
      <input
        className="filter-input"
        onChange={[Function]}
        placeholder="Search..."
      />
      <div
        className="select-all-visible"
      >
        <a
          onClick={[Function]}
        >
          Clear list
        </a>
      </div>
      <ul
        className="ms-list"
      >
        <li
          className=""
          onClick={[Function]}
        >
          1
        </li>
        <li
          className="group-label"
          onClick={[Function]}
        >
          Title 2
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          7
        </li>
      </ul>
    </div>
  </div>
</div>
`;
